wall_e/src/resources/cogs/ban.py

```diff
@@ -0,0 +1,426 @@
+                    user_id = banned.target.id
+                    mod = banned.user.name + '#' + banned.user.discriminator
+                    mod_id = banned.user.id
+                    ban_date = banned.created_at.timestamp()
+                    reason = banned.reason if banned.reason else 'No Reason Given!'
+
+                else:
+                    username = ban.user.name + '#' + ban.user.discriminator
+                    user_id = ban.user.id
+
+                ban_records.append(BanRecords(
+                    username=username,
+                    user_id=user_id,
+                    mod=mod,
+                    mod_id=mod_id,
+                    ban_date=ban_date,
+                    reason=reason
+                ))
+
+        await BanRecords.insert_records(ban_records)
+
+        await ctx.send(f"Moved `{len(ban_records)}` active bans from guild bans to walle bans.")
+        logger.info(f"[Ban convertbans()] total of {len(ban_records)} bans moved into walle ban system")
+
+    @commands.command()
+    async def ban(self, ctx, user: discord.Member, purge_window_days: int = 1, *reason: str):
+        """Bans a user from the guild"""
+
+        logger.info(f"[Ban ban()] Ban command detected from {ctx.author} with args: user={user}, reason={reason}, " +
+                    f"purge_window_days={purge_window_days}]")
+
+        # confirm at least 1 @ mention of user to ban
+        if len(ctx.message.mentions) < 1:
+            logger.info("[Ban ban()] No user were @ mentioned in the args")
+            e_obj = await em(ctx=ctx, title="Invalid Arguments",
+                             content=[("Error", "Please @ mention the user to ban"),
+                                      ("Command Usage", "`.ban @user [<# of days to purge messages>] [<reason>]`"),
+                                      ("Example Usage", "`.ban @user1 2 they're being weird`")],
+                             colour=self.error_colour,
+                             footer="Command Error")
+            if e_obj:
+                await ctx.send(embed=e_obj)
+            return
+
+        logger.info(f"[Ban ban()] User to ban: {user}")
+
+        # construct reason
+        reason = ' '.join(reason)
+        reason = reason if reason else "No Reason Given."
+        logger.info(f"[Ban ban()] Ban reason{reason}")
+
+        # ban
+        dm = True
+        # for user in users_to_ban:
+        ban = BanRecords(
+                         username=user.name + '#' + user.discriminator,
+                         user_id=user.id,
+                         mod=ctx.author.name+'#'+ctx.author.discriminator,
+                         mod_id=ctx.author.id,
+                         reason=reason
+                         )
+
+        logger.info(f"[Ban ban()] Banning {ban.username} with id {ban.user_id}")
+
+        # add to ban_list
+        self.ban_list.append(ban.user_id)
+
+        # dm banned user
+        e_obj = discord.Embed(title="Ban Notification",
```

Comment:
same question here

Reply:
I think this one can be used by the embed utility though? @TitanVJ

Reply:
icon_url in footer not supported in the utility function